### ISSUE TITLE
TS 3.8 `import type` support

### ISSUE THREAD

**Issue by merrywhether** [relay-tools/relay-compiler-language-typescript#177]
With the release of TS 3.8, the type imports in the generated files should switch from the style
```
import { ConcreteRequest } from "relay-runtime";
```
to
```
import type { ConcreteRequest } from "relay-runtime";
```
The is more semantically correct, and also allows TS-Relay users to enable the `--importsNotUsedAsValues error` option to enable stricter type usage (currently they would see `This import is never used as a value and must use 'import type' because the 'importsNotUsedAsValues' is set to 'error'.`).

Optionally, if you don't want to force TS 3.8 usage with this library, another option to get around it would be:
```
// @ts-ignore
import { ConcreteRequest } from "relay-runtime";
```
which is less elegant but more backwards compatible.

**Comment by alloy** [comment]
I don’t think we should require TS 3.8 just yet. I’m definitely open to merging a PR to allow enabling stricter flags, though! :+1:

**Comment by sibelius** [comment]
can `import type` be a config?

**Comment by alloy** [comment]
There’s no easy way to add config for just the plugin and I think having less config is better UX anyways.

**Comment by renanmav** [comment]
We can’t pass options [here](https://github.com/facebook/relay/blob/master/packages/relay-compiler/language/RelayLanguagePluginInterface.js#L280) and [here](https://github.com/facebook/relay/blob/master/packages/relay-compiler/language/RelayLanguagePluginInterface.js#L240)?

**Comment by alloy** [comment]
I mean, we can technically probably always find a way, but I prefer to not even have options if there isn’t a great need. Introducing options isn’t free, it increases complexity of a codebase and the ability of users to understand how to use something.

**Comment by renanmav** [comment]
Hm, so I’m not sure what you meant by "allow enabling stricter flags". Could you elaborate it a bit more?

**Comment by alloy** [comment]
I mean the `—importsNotUsedAsValues` Option that OP mentioned.

**Comment by merrywhether** [comment]
Any reason that the `// @ts-ignore` path before the import (and also for other things like `ReaderFragment`, `FragmentRefs`, etc) wouldn't work then? From playing around with my generated files, TS still properly processes those types it just doesn't alert on those lines in the file. This would be similar to the already-present disabling of `tslint` and `eslint`.

**Comment by alloy** [comment]
@merrywhether Nope, I think using the magic comments is a good solution for now 👍

**Comment by alloy** [comment]
:rocket: Issue was released in v13.0.1 :rocket:

**Comment by alloy** [comment]
:rocket: Issue was released in v13.0.1 :rocket:

**Comment by merrywhether** [comment]
Verifying this works in the latest version. Thanks!
